File: project/app/src/main/java/org/owntracks/android/injection/modules/android/FragmentModules/FinishFragmentModule.java
package org.owntracks.android.injection.modules.android.FragmentModules;


import androidx.fragment.app.Fragment;

import org.owntracks.android.injection.scopes.PerFragment;
import org.owntracks.android.ui.welcome.finish.FinishFragment;

import dagger.Binds;
import dagger.Module;

@Module(includes = BaseSupportFragmentModule.class)
public abstract class FinishFragmentModule {

    @Binds
    @PerFragment
    abstract Fragment bindSupportFragment(FinishFragment f);
}

File: project/app/src/main/java/org/owntracks/android/ui/welcome/finish/FinishFragmentModule.kt
package org.owntracks.android.ui.welcome.finish

import androidx.fragment.app.Fragment
import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.FragmentComponent
import dagger.hilt.android.scopes.FragmentScoped

@InstallIn(FragmentComponent::class)
@Module
abstract class FinishFragmentModule {
    @Binds
    @FragmentScoped
    abstract fun bindSupportFragment(f: FinishFragment?): Fragment?
}